### PR TITLE
lnwallet: use new errors returned from `rpcclient`

### PR DESCRIPTION
**Pull request by yyforyongyu**
Depends on,
- https://github.com/btcsuite/btcd/pull/2138
- https://github.com/btcsuite/btcwallet/pull/918

This PR starts to make use of the detailed RPC error types, enabling us to have better control over RBF failures.

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling in `PublishTransaction` function using `btcd/rpcclient`.
	- Added custom callbacks for healthcheck events.
- **Refactor**
	- Improved error mapping in `PublishTransaction` for clarity.
	- Updated `broadcastErrorMapper` to use `rpcclient.MapRPCErr`.
- **Tests**
	- Adjusted test scenarios for transaction rejections with RBF settings.
- **Chores**
	- Streamlined build process by removing a debug log argument.
- **Documentation**
	- Updated release notes to reflect error handling and healthcheck changes.

### CODE REVIEW

File: lnwallet/btcwallet/btcwallet.go [hunk 1]
```diff
@@ -1213,47 +1212,42 @@ func (b *BtcWallet) ListUnspentWitness(minConfs, maxConfs int32,
 	return witnessOutputs, nil
 }
 
+// mapRpcclientError maps an error from the rpcclient package to defined error
```
Comment (ziggie1984) [inline lnwallet/btcwallet/btcwallet.go:1215]: Nit: Maybe add that we are mapping the bitcoind errors from the `sendrawtransaction` command not the one from `testmempoolaccept` ?

File: lnwallet/btcwallet/btcwallet.go [hunk 2]
```diff
@@ -1319,15 +1313,15 @@ func (b *BtcWallet) PublishTransaction(tx *wire.MsgTx, label string) error {
 	// `PublishTransaction` again because we need to mark the label in the
 	// wallet. We can remove this exception once we have the above TODO
 	// fixed.
-	case lnutils.ErrorAs[*base.ErrInMempool](err),
-		lnutils.ErrorAs[*base.ErrAlreadyConfirmed](err):
+	case errors.Is(err, rpcclient.ErrTxAlreadyInMempool),
```
Comment (ziggie1984) [inline lnwallet/btcwallet/btcwallet.go:1316]: Nit: ` we'd always get ErrInMempool` is outdated above in the comment section.

Moreover I wonder why we mix here errors from the `testmempoolaccept` and `sendrawtransaction` cmd, don't we just have the former here as an errorr ?

File: lnwallet/test/test_interface.go [hunk 3]
```diff
@@ -1819,16 +1819,27 @@ func testPublishTransaction(r *rpctest.Harness,
...
-			require.ErrorIs(t, err, lnwallet.ErrDoubleSpend)
+
+			// If RBF is not enabled, we expect this to be rejected
+			// because it is a double spend.
+			expectedErr := lnwallet.ErrDoubleSpend
+
+			// If RBF is enabled, we expect it to be rejected
+			// because it doesn't pay enough fees.
+			if rbf {
+				expectedErr = rpcclient.ErrInsufficientFee
```
Comment (ziggie1984) [inline lnwallet/test/test_interface.go:1838]: Q: wondering whether we should get rid of the internal LND mapping or add this error to the mapping ?

**Review by ziggie1984 — COMMENTED** [review]
Looks very good had just some Nits.

I wonder if we map BTCD errors resulting from the `testmempoolaccept` cmd to bitcoind's `sendrawtransaction` errors or are we strict and try to always map them equally:

btcd's testmempool accept errors => bitcoind's testmempool accept errors ?

File: lnwallet/btcwallet/btcwallet.go (hunk 1 above)
Comment (yyforyongyu) [reply lnwallet/btcwallet/btcwallet.go:1215]: we are mapping the reject reason from `testmempoolaccept` tho, updated the comments.

File: lnwallet/btcwallet/btcwallet.go (hunk 2 above)
Comment (yyforyongyu) [reply lnwallet/btcwallet/btcwallet.go:1316]: Updated the docs!

When an error is returned from `testmempoolaccept`, it's returned directly without any mapping. Here we just map the reject reason, and weirdly bitcoind gives different error messages for the same error case when calling `testmempoolaccept` vs `sendrawtransaction`.

File: lnwallet/test/test_interface.go (hunk 3 above)
Comment (yyforyongyu) [reply lnwallet/test/test_interface.go:1838]: I'd like to get rid of the internal LND mapping and expose the detailed RPC errors to the callers - we cannot do it yet tho, as I think there would be compatibility issues.

**Review by guggero — APPROVED** [review]
LGTM :tada:

File: lnwallet/btcwallet/btcwallet.go (hunk 2 above)
Comment (ziggie1984) [reply lnwallet/btcwallet/btcwallet.go:1297]: Ok in the `btcd` package you reference the error `ErrTxAlreadyConfirmed` to origin from `sendrawtransaction` so do you really need to check for it here when we only test for `testmempoolaccept ` ?

Comment (yyforyongyu) [reply lnwallet/btcwallet/btcwallet.go:1297]: oh that's what you meant, yeah we don't need to check it, updated. And for references:
https://github.com/bitcoin/bitcoin/blob/3d216baf91ca754e46e89788205513a956ec6d0a/test/functional/rpc_rawtransaction.py#L438-L439

Comment (yyforyongyu) [reply lnwallet/btcwallet/btcwallet.go:1297]: actually we still need this error mapping as `btcd` would return it here.

Comment (ziggie1984) [reply lnwallet/btcwallet/btcwallet.go:1297]: ohh you are right, thank you

**Review by ziggie1984 — APPROVED** [review]
LGTM

**Pull request merged**